### PR TITLE
[Features] translation, donors, donations, subscriptions queries, inivitation and welcome feature

### PR DESCRIPTION
**Pull request by jstaricco10**

### CODE REVIEW

File: prisma/seed/donation/factory.ts
Comment (jopealgorta) [inline prisma/seed/donation/factory.ts:1]: This table I think should not de seeded. This model also will change a lot in the next days while testing deeper the MP API. Also, I think this type of models are not worth seeding.

File: prisma/seed/donor/factory.ts
Comment (jopealgorta) [inline prisma/seed/donor/factory.ts:1]: Same here.

File: prisma/seed/subscription/factory.ts
Comment (jopealgorta) [inline prisma/seed/subscription/factory.ts:1]: Same here.

File: src/api/donations/dto/create-donation.input.ts [hunk 1]
```diff
@@ -2,6 +2,9 @@ import { InputType, Field, Int } from '@nestjs/graphql';
 
 @InputType()
 export class CreateDonationInput {
+  @Field(() => Int)
```
Comment (jopealgorta) [inline src/api/donations/dto/create-donation.input.ts]: This field should not be here. The donor is more like an internal model to normalize it from donations and subscriptions. A user (using the api) will never create a donor and then a donation.

File: src/api/donors/donors.resolver.ts [hunk 2]
```diff
@@ -22,6 +23,21 @@ export class DonorsResolver {
     return this.donorsService.findOne(id);
   }
 
+  @Query(() => DonorPagination, { name: 'donorsByOrganization' })
```
Comment (jopealgorta) [inline src/api/donors/donors.resolver.ts]: I don't think we need these ones. Remember, donors are only associated with donations, not separately.

File: src/api/donors/donors.service.ts [hunk 3]
```diff
@@ -26,6 +27,59 @@ export class DonorsService {
     return this.prisma.donor.findUnique({ where: { id } });
   }
 
+  async findAllByProjectId(projectId: number, args: PaginationArgs = { page: 1, itemsPerPage: 5, filter: '' }) {
+    const projectUsers = await this.prisma.projectUser.findMany({ where: { projectId } });
+    const userIds = projectUsers.map((projectUser) => projectUser.userId);
+
+    const projectDonors = await this.prisma.donor.findMany({
```
Comment (jopealgorta) [inline src/api/donors/donors.service.ts:34]: Project Users and donors have nothing to do with each other. The project user are the members of the project, and the donors are those who made a economic donation.

File: src/api/donors/entities/donor.entity.ts [hunk 4]
```diff
@@ -18,3 +19,12 @@ export class Donor extends BaseEntity {
...
   paymentMethod?: string;
 }
+
+@ObjectType()
+export class DonorPagination {
+  @Field(() => [Donor])
+  donors: [Donor];
+
+  @Field(() => Int)
+  total: [User];
```
Comment (jopealgorta) [inline src/api/donors/entities/donor.entity.ts:29]:
```suggestion
  total: number;
```

File: src/api/donors/entities/donor.entity.ts [hunk 5]
```diff
@@ -1,5 +1,6 @@
-import { ObjectType, Field } from '@nestjs/graphql';
+import { ObjectType, Field, Int } from '@nestjs/graphql';
 import { BaseEntity } from '../../common/entities/base.entity';
+import { User } from 'src/api/users/entities/user.entity';
```
Comment (jopealgorta) [inline src/api/donors/entities/donor.entity.ts:3]: This is not needed.

File: src/api/donations/entities/donation.entity.ts [hunk 6]
```diff
@@ -31,3 +32,12 @@ export class Donation extends BaseDonation {
...
   project: Project;
 }
+
+@ObjectType()
+export class DonationPagination {
+  @Field(() => [Donation])
+  donations: [Donation];
+
+  @Field(() => Int)
+  total: [User];
```
Comment (jopealgorta) [inline src/api/donations/entities/donation.entity.ts:42]:
```suggestion
  total: number;
```

File: src/api/projects/entities/project.entity.ts [hunk 7]
```diff
@@ -52,6 +52,9 @@ export class Project extends BaseEntity {
 
   @Field(() => Organization, { nullable: true })
   organization: Organization;
+
+  @Field(() => String, { nullable: true })
```
Comment (jopealgorta) [inline src/api/projects/entities/project.entity.ts]: If we place `{ nullable: true }` then the field must have the `?` like `slug?:`. Anyways, this field is required.

File: src/api/projects/projects.service.ts [hunk 8]
```diff
@@ -71,6 +71,19 @@ export class ProjectsService {
     return project;
   }
 
+  public async findOneBySlug(slug: string, user: User) {
+    const project = await this.prisma.project.findFirst({ where: { slug } });
+    if (!project) throw new NotFoundError('Project not found');
+    if (!user || user.role === Role.USER || project.organizationId !== user.organizationId) {
+      delete project.mpAccessToken;
+      delete project.mpPublicKey;
```
Comment (jopealgorta) [inline src/api/projects/projects.service.ts]: I remove the mpPulicKey from the excluded fields, as could be used in the FE. Could you remove it here too?

File: src/api/subscriptions/dto/create-subscription.input.ts [hunk 9]
```diff
@@ -3,6 +3,9 @@ import { FrequencyInterval } from '@prisma/client';
 
 @InputType()
 export class CreateSubscriptionInput {
+  @Field(() => Int)
```
Comment (jopealgorta) [inline src/api/subscriptions/dto/create-subscription.input.ts]: Not needed.

File: src/api/subscriptions/entities/subscription.entity.ts [hunk 10]
```diff
@@ -43,3 +44,12 @@ export class Subscription extends BaseSubscription {
...
   name: 'FrequencyInterval',
 });
+
+@ObjectType()
+export class SubscriptionPagination {
+  @Field(() => [Subscription])
+  subscriptions: [Subscription];
+
+  @Field(() => Int)
+  total: [User];
```
Comment (jopealgorta) [inline src/api/subscriptions/entities/subscription.entity.ts]:
```suggestion
  total: number;
```

File: src/api/users/entities/user.entity.ts [hunk 11]
```diff
@@ -26,3 +26,12 @@ export class User extends BaseEntity {
...
   name: 'Role',
 });
+
+@ObjectType()
+export class OrgAdminPagination {
+  @Field(() => [User])
+  admins: [User];
+
+  @Field(() => Int)
+  total: [User];
```
Comment (jopealgorta) [inline src/api/users/entities/user.entity.ts]:
```suggestion
  total: number;
```

**Review by jopealgorta — COMMENTED** [review]
Great Juanma! Thanks for addressing this for me.

File: prisma/seed/donation/factory.ts
Comment (jstaricco10) [reply prisma/seed/donation/factory.ts:1]: I already did and its pretty easy to change so I think I'll leave it for now

File: src/api/subscriptions/dto/create-subscription.input.ts (hunk 9 above)
Comment (jstaricco10) [reply src/api/subscriptions/dto/create-subscription.input.ts]: What is not needed?

Comment (jstaricco10) [reply src/api/subscriptions/dto/create-subscription.input.ts]: I see, same as for donations

File: src/api/projects/projects.service.ts [hunk 12]
```diff
@@ -64,7 +64,6 @@ export class ProjectsService {
 
     if (!project) throw new NotFoundError('Project not found');
     if (!user || user.role === Role.USER || project.organizationId !== user.organizationId) {
-      delete project.mpAccessToken;
```
Comment (jopealgorta) [inline src/api/projects/projects.service.ts:67]: It was the public key!!!

**Review by jopealgorta — APPROVED** [review]

**Pull request merged**